src/test/java/ibm/jceplus/junit/base/BaseTestEdDSASignatureInterop.java

```diff
@@ -81,8 +86,15 @@ private byte[] doSign(String sigAlgo, byte[] message, PrivateKey privateKey) thr
     //Verify the message with BouncyCastle provided Ed25519
     protected void doVerifyEd25519(byte[] message, byte[] signedBytes, PublicKey publicKey)
             throws Exception {
```

Comment:
This code block along with others in this file look quite different. Could you explain where this code came from?

Reply:
The `com.ibm.crypto.plus.provider.EdDSAPublicKeyImpl` was previously used, which shouldn't be public. So, I the `EdDSAKeyFactory` to get the spec, which is allowed, and then used the inverse process used in the constructor of the `EdDSAPublicKeyImpl` to extract the information required by `BC`'s `Ed25519PublicKeyParameters`, without exposing any unnecessary classes or methods.